core/package.json

```diff
@@ -99,7 +100,8 @@
     "test.e2e.docker": "npm run docker.build && docker run -it --rm -e DISPLAY=$(cat docker-display.txt) -v $(cat docker-display-volume.txt) --ipc=host --mount=type=bind,source=./,target=/ionic ionic-playwright npm run test.e2e --",
     "test.e2e.docker.update-snapshots": "npm run test.e2e.docker -- --update-snapshots",
     "test.e2e.docker.ci": "npm run docker.build && docker run -e CI='true' --rm --ipc=host --mount=type=bind,source=./,target=/ionic ionic-playwright npm run test.e2e --",
-    "test.report": "npx playwright show-report"
+    "test.report": "npx playwright show-report",
+    "tokens": "node ./tokens.js && npm run lint.sass.fix"
```

Comment:
This script should probably be run on build. My main concern here is what happens if someone modifies the JSON file but forgets to run this script? We'd potentially ship with outdated styles.

I'd recommend doing something like this:

1. Update the `build` script above to call `npm run tokens` (You could also rename this to `build.tokens` if you wanted naming consistency). This ensures that the latest tokens are generated on `npm run build`. Note that this should be run **before** `npm run build.css` since the tokens script outputs scss.
2. In the [`build-core` workflow](https://github.com/ionic-team/ionic-framework/blob/b5cb828861c568919a8a8f1090677f37dc1ce424/.github/workflows/actions/build-core/action.yml#L34)  and the [`build-core-stencil-prerelease` workflow](https://github.com/ionic-team/ionic-framework/blob/b5cb828861c568919a8a8f1090677f37dc1ce424/.github/workflows/actions/build-core-stencil-prerelease/action.yml#L32) make sure the `core/src/foundations` directory is cached.

The benefit of this is if anyone forgets to build the latest scss file the [`test-core-clean-build`](https://github.com/ionic-team/ionic-framework/blob/main/.github/workflows/actions/test-core-clean-build/action.yml) will fail in CI.

Reply:
Hum thank you for this feedback, I think it makes total sense 👍 

1. Already updated the script name and move it to the build command.
2. Added the `core/src/foundations` to the path attribute of those actions. I don't have much context about this, is there anything more I should do?

Reply:
Looks great! I tested this out on a separate branch, and CI correctly fails if you change the source json file without re-running the build: https://github.com/ionic-team/ionic-framework/actions/runs/8560100688/job/23458371826?pr=29276